src/webhooks.ts

```diff
@@ -1,14 +1,14 @@
 import { Octokit } from "@octokit/core";
 import { createAppAuth } from "@octokit/auth-app";
 import { createUnauthenticatedAuth } from "@octokit/auth-unauthenticated";
-import { Webhooks } from "@octokit/webhooks";
+import { Webhooks, EmitterWebhookEvent } from "@octokit/webhooks";
 
 import { Options } from "./types";
 
 export function webhooks(
   appOctokit: Octokit,
   options: Required<Options>["webhooks"]
-) {
+): Webhooks<EmitterWebhookEvent, EmitterWebhookEvent & { octokit: Octokit }> {
```

Comment:
I was hoping that we won't need to set the Type parameters anymore with v8 😕

Reply:
Nice. I was trying to make it compile in my local but I was missing this second generic argument part.

---

I would like to find a way to let TypeScript auto-infer the type yes... @G-Rath do you think it would be possible (in a different PR) to get this `BaseWebhookEvent` inference properly solved?

Reply:
The solution would be to export `BaseWebhookEvent`, which in itself might not be the worst idea but for this specific case is: the resulting type will be every possible element of the union, which is huge.

Instead, providing the type explicitly is the better solution, as it's for a public export - relying on inference means it'd be possible for the types to subtly change without knowing (even via say a package update if we were to change the types upstream).

Reply:
> relying on inference means it'd be possible for the types to subtly change without knowing (even via say a package update if we were to change the types upstream).

that would also include fixes though, right? 

> providing the type explicitly is the better solution

as it is right now in this PR? Or would you change anything?

Reply:
> that would also include fixes though, right?

Yes, but actually you'll never get fixes, as I just realised this'll never actually be inferred: type definitions have to be explicitly typed in full, as they don't have the source code which is required for TypeScript to infer the code.

i.e

```
export const fn = () => 'hello world!';
```

will always give you:

```
export const fn: () => string;
```

-------

So for any "fixes" we'd always have to publish a new version for the type definitions to include them.
If we rely on inference, we won't know that the types need to be updated, since it won't cause an error.

Reply:
> as it is right now in this PR? Or would you change anything?

It looks good to me as is. I suspect it'll need changing when <that PR I was just looking at whose tab I've now closed> is merged